### PR TITLE
fix: Testcafe using Posthog cloud

### PR DESCRIPTION
**Pull request by benjackwhite**
## Changes

Simplifies and fixes E2E tests with testcafe by using PostHog cloud

## Checklist
- [x] Tests for new code (see [advice on the tests we use](https://github.com/PostHog/posthog-js#tiers-of-testing))
- [x] Accounted for the impact of any changes across different browsers
- [x] TypeScript definitions (module.d.ts) updated and in sync with library exports (if applicable)

### CODE REVIEW

File: .github/workflows/testcafe.yml [hunk 1]
```diff
@@ -3,82 +3,30 @@ on: [pull_request]
...
-  SECRET_KEY: 'abcdef' # unsafe - for testing only
-  DATABASE_URL: 'postgres://posthog:posthog@localhost:5432/test_posthog'
-  REDIS_URL: 'redis://localhost'
-  TEST: 'true'
+  POSTHOG_PROJECT_KEY: "${{ secrets.POSTHOG_PROJECT_KEY }}"
+  POSTHOG_API_KEY: "${{ secrets.POSTHOG_API_KEY }}"
   BROWSERSTACK_DEBUG: 'true'
   BROWSERSTACK_NETWORK_LOGS: 'true'
   BROWSERSTACK_CONSOLE: 'info'
-  PLUGIN_SERVER_INGESTION: 'true'
```
Comment (pauldambra) [inline .github/workflows/testcafe.yml:13]: yes! tidy up!

File: testcafe/helpers.js [hunk 2]
```diff
@@ -3,7 +3,13 @@ import path from 'path'
 import { RequestLogger, RequestMock, ClientFunction } from 'testcafe'
 import fetch from 'node-fetch'
 
-const HEADERS = { Authorization: 'Bearer e2e_demo_api_key' }
+// NOTE: These tests are run against a dedicated test project in PostHog cloud
```
Comment (pauldambra) [inline testcafe/helpers.js:6]: 🙌

File: testcafe/helpers.js [hunk 3]
```diff
@@ -51,15 +67,13 @@ export async function retryUntilResults(operation, target_results, limit = 100)
...
-export async function queryAPI() {
-    const response = await fetch('http://localhost:8000/api/event', {
+export async function queryAPI(testSessionId) {
+    const url = `${POSTHOG_API_HOST}/api/projects/${POSTHOG_API_PROJECT}/events?properties=[{"key":"testSessionId","value":["${testSessionId}"],"operator":"exact","type":"event"}]`
+    const response = await fetch(url, {
         headers: HEADERS,
     })
 
-    const { results } = JSON.parse(await response.text())
+    const data = await response.text()
```
Comment (pauldambra) [inline testcafe/helpers.js:76]: Should we explode if `!response.ok` so future errors are clearer?

**Review by pauldambra — APPROVED** [review]
Supersedes and closes #414 

I love the simplicity of this approach compared to spinning up the stack 💖

File: testcafe/helpers.js (hunk 3 above)
Comment (benjackwhite) [reply testcafe/helpers.js:76]: That's a good idea. I'll add it in

**Pull request merged**